### ISSUE TITLE
global_sort_pool device error before and after

### ISSUE THREAD

**Issue by leftthomas** [rusty1s/pytorch_geometric#80]
@rusty1s when I use `global_sort_pool` in my model, I debug it and found something wrong, before this function, the data is on cuda, but after this function, the returned value is not on cuda.
<img width="764" alt="qq20190126-190055 2x" src="https://user-images.githubusercontent.com/9991443/51786271-c679da00-219c-11e9-87ae-052dfd4bc300.png">
Further more, this bug is only occur when I use `TUDataset` and the data is `MUTAG`, but when data is `DD`, this bug is not occur

**Comment by leftthomas** [comment]
@rusty1s Here is an example to reproduce this error. 
```python
import argparse
from torch.optim import Adam
from torch_geometric.data import DataLoader
from torch_geometric.datasets import TUDataset
from tqdm import tqdm
import torch
from torch import nn
from torch_geometric.nn import GCNConv, global_sort_pool


class Model(nn.Module):
    def __init__(self, num_features, num_classes):
        super(Model, self).__init__()

        self.conv1 = GCNConv(num_features, 32)
        self.conv2 = GCNConv(32, 32)
        self.conv3 = GCNConv(32, 32)
        self.classifier = nn.Linear(32*50*3, num_classes)

    def forward(self, data):
        x, edge_index, batch = data.x, data.edge_index, data.batch

        x_1 = torch.tanh(self.conv1(x, edge_index))
        x_2 = torch.tanh(self.conv2(x_1, edge_index))
        x_3 = torch.tanh(self.conv3(x_2, edge_index))
        x = torch.cat([x_1, x_2, x_3], dim=-1)
        x = global_sort_pool(x, batch, k=50)
        x = x.view(x.size(0), -1)
        out = self.classifier(x)

        return torch.sigmoid(out)


if __name__ == '__main__':

    parser = argparse.ArgumentParser(description='Train Model')
    parser.add_argument('--data_type', default='DD', type=str,
                        choices=['DD', 'REDDIT-BINARY', 'REDDIT-MULTI-5K', 'REDDIT-MULTI-12K', 'PTC_MR', 'NCI1',
                                 'NCI109', 'PROTEINS', 'IMDB-BINARY', 'IMDB-MULTI', 'MUTAG', 'ENZYMES', 'COLLAB'],
                        help='dataset type')
    parser.add_argument('--batch_size', default=20, type=int, help='train batch size')

    opt = parser.parse_args()

    DATA_TYPE = opt.data_type
    BATCH_SIZE = opt.batch_size

    data_set = TUDataset(DATA_TYPE, DATA_TYPE).shuffle()
    NUM_FEATURES, NUM_CLASSES = data_set.num_features, data_set.num_classes
    # create a 90/10 train/test split
    train_len = int(0.9 * len(data_set))
    train_set, test_set = data_set[:train_len], data_set[train_len:]
    train_loader = DataLoader(dataset=train_set, batch_size=BATCH_SIZE, shuffle=True)
    test_loader = DataLoader(dataset=test_set, batch_size=BATCH_SIZE, shuffle=False)

    print('# %s details:' % data_set, '[train] ', len(train_set), '[test] ', len(test_set), '[num_classes] ',
          data_set.num_classes)

    model = Model(NUM_FEATURES, NUM_CLASSES)
    loss_criterion = nn.CrossEntropyLoss()
    if torch.cuda.is_available():
        model = model.to('cuda')

    optimizer = Adam(model.parameters())
    model.train()
    for data in tqdm(train_loader):
        labels = data.y
        if torch.cuda.is_available():
            data, labels = data.to('cuda'), labels.to('cuda')
        optimizer.zero_grad()
        classes = model(data)
        loss = loss_criterion(classes, labels)
        optimizer.step()
```
And I found something strange, according to the [official website](https://ls11-www.cs.tu-dortmund.de/staff/morris/graphkerneldatasets) about these dataset, the `REDDIT-BINARY` dataset should have 2 classes, but when I use `TUDataset` to load this data, the `num_classes`property of it is 3. The same error is occurred on `PTC_MR` and `MUTAG`.

**Comment by rusty1s** [comment]
Wow, really very watchful. I fixed both bugs in current master.

Regarding `global_sort_pool`: the error only occurred when there was no graph in the dataset with a larger number of nodes than `k`.

Regarding the TUDatasets: I didn't know that class labels for binary classifications are encoded as `-1` and `1` :(

Thank you so much and sorry for the inconvenience!
